pkg/bpf/map.go

```diff
@@ -439,25 +496,73 @@ func (m *Map) Lookup(key MapKey) (MapValue, error) {
 }
 
 func (m *Map) Update(key MapKey, value MapValue) error {
+	var err error
+
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
-	if err := m.Open(); err != nil {
+	defer func() {
+		if m.cache == nil {
+			return
+		}
+
+		desiredAction := OK
+		if err != nil {
+			desiredAction = Insert
+		}
+
+		m.cache[key.String()] = &cacheEntry{
+			Key:           key,
+			Value:         value,
+			DesiredAction: desiredAction,
+			LastError:     err,
+		}
+	}()
+
+	if err = m.Open(); err != nil {
 		return err
 	}
 
-	return UpdateElement(m.fd, key.GetKeyPtr(), value.GetValuePtr(), 0)
+	err = UpdateElement(m.fd, key.GetKeyPtr(), value.GetValuePtr(), 0)
+	return err
+}
+
+func (m *Map) deleteCacheEntry(key MapKey, err error) {
```

Comment:
Add a comment that specifies that this assumes the map's mutex is being held.

Reply:
Fixed